nodejs/awsx/apigateway/api.ts

```diff
@@ -118,17 +118,51 @@ function isStaticRoute(route: Route): route is StaticRoute {
 }
 
 /**
- * An apigateway route that maps to some target uri, or some elastic-load-balancer host/port.
+ * An apigateway route for an integration.
+ * https://docs.aws.amazon.com/apigateway/api-reference/resource/integration/ for more details.
  */
-export interface ProxyRoute {
+export interface IntegrationRoute {
     path: string;
-    target: pulumi.Input<ProxyTarget> | ProxyRouteTargetProvider;
+    target: pulumi.Input<IntegrationTarget> | IntegrationRouteTargetProvider;
 }
 
+export type IntegrationConnectionType = "INTERNET" | "VPC_LINK";
+export type IntegrationType = "aws" | "aws_proxy" | "http" | "http_proxy" | "mock";
+export type IntegrationPassthroughBehavior = "when_no_match" | "when_no_templates" | "never";
+
 /**
  * See https://docs.aws.amazon.com/apigateway/api-reference/resource/integration/ for more details.
  */
-export interface ProxyTarget {
+export interface IntegrationTarget {
+    /**
+     * Specifies an API method integration type. The valid value is one of the following:
+     *
+     * aws: for integrating the API method request with an AWS service action, including the Lambda
+     * function-invoking action. With the Lambda function-invoking action, this is referred to as
+     * the Lambda custom integration. With any other AWS service action, this is known as AWS
+     * integration.
+     *
+     * aws_proxy: for integrating the API method request with the Lambda function-invoking action
+     * with the client request passed through as-is. This integration is also referred to as the
+     * Lambda proxy integration.
+     *
+     * http: for integrating the API method request with an HTTP endpoint, including a private HTTP
+     * endpoint within a VPC. This integration is also referred to as the HTTP custom integration.
+     *
+     * http_proxy: for integrating the API method request with an HTTP endpoint, including a private
+     * HTTP endpoint within a VPC, with the client request passed through as-is. This is also
+     * referred to as the HTTP proxy integration.
+     *
+     * mock: for integrating the API method request with API Gateway as a "loop-back" endpoint
+     * without invoking any backend.
+     */
+    type: pulumi.Input<IntegrationType>;
+
+    /**
+     * Specifies the integration's HTTP method type.  Currently, the only supported type is 'ANY'.
+     */
+    httpMethod?: "ANY";
```

Comment:
note: this is a limitation to how we currently set up swagger paths (and compute sha hashes and whatnot).  we can relax this in the future, but it will require a bit more effort.